app/controllers/spree/admin/gdpr_requests_controller.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module Spree
+  module Admin
+    class GdprRequestsController < ResourceController
+      before_action :load_data
+
+      def index
+        respond_with(@collection)
+      end
+
+      def serve
+        unless @object
+          redirect_to admin_gdpr_requests_path, flash: { error: t('gdpr.exports.error') }
+          return
+        end
+
+        @object.serve
+        redirect_to admin_gdpr_requests_path, notice: t('gdpr.exports.scheduled')
+      end
+
+      private
+
+      def collection
+        @collection ||=
+          Spree::GdprRequest
+            .order('served_at IS NOT NULL', created_at: :desc)
+            .page(params[:page] || 0)
+      end
+
+      def load_data
+        @users = Spree::User.order(:email)
+        @intents = Spree::GdprRequest.intents.collect do |slug, _index|
+          [slug.split('_').map(&:capitalize).join(' '), slug]
```

Comment:
Since you also have the same logic in a view, perhaps you can extract it into a helper?